tests/hw18/hw18.spec.js

```diff
@@ -0,0 +1,242 @@
+const testPassword = 'Qwertyui1';
+const invalidName = 'test123';
+const invalidLastName = 'test123';
+const shortName = 't';
+const longName = 'test123test123test123';
+const minimumLengthName = 'te';
+const maximumLengthName = 'test123test123test12';
+const shortLastName = 't';
+const minimumLengthLastName = 'te';
+const maximumLengthLastName = 'test123test123test12';
+const longLastName = 'test123test123test123';
+const invalidEmails = ['test', 'test@', 'test@test', 'test@test.'];
+const invalidPasswords = ['qwertyu', 'Qwerty1', 'Qwertyui', 'QwertyuiQwertyu1'];
+
+test.describe('Sign Up Positive Flow', ()=>{
+    
+    test.beforeEach(async ({ page })=>{
+        await page.goto(''); 
+    });
+
+    // Delete test user after each successful sign up
+    test.afterEach(async ({page})=>{
+        const settingsBtn = page.getByRole('link', { name: 'Settings' });
+        const removeAccountBtn = page.getByRole('button', { name: 'Remove my account' });
+        const confirmRemoveAccountBtn = page.getByRole('button', { name: 'Remove' });
+        await settingsBtn.click();
+        await removeAccountBtn.click();
+        await confirmRemoveAccountBtn.click();
+    });
+
+    test('Should be successfully signed up', async ({page})=>{
+        
+        const signUpBtn = page.getByRole('button', { name: 'Sign up' });
+        const nameInput =  page.locator('#signupName');
+        const lastNameInput = page.locator('#signupLastName');
+        const emailInput = page.locator('#signupEmail');
+        const passwordInput = page.getByLabel('Password', { exact: true });
+        const reenterPasswordInput = page.getByLabel('Re-enter password');
+        const registerBtn = page.getByRole('button', { name: 'Register' });
+        const myProfileBtn = page.getByRole('button', { name: 'My profile' });
+
+        await signUpBtn.click();
+        await nameInput.fill(testName);
+        await lastNameInput.fill(testLastName);
+        await emailInput.fill(testEmail);
+        await passwordInput.fill(testPassword);
+        await reenterPasswordInput.fill(testPassword);
+        await registerBtn.click();
+
+        await expect(myProfileBtn).toBeVisible();
+    });
+});
+
+test.describe('Input fields validation', ()=>{
+    
+    test.beforeEach(async ({page})=>{
+        await page.goto('');
+        const signUpBtn = page.getByRole('button', { name: 'Sign up' });
+        await signUpBtn.click();
+    });
+
+    test.afterEach(async ({page})=> {
+        const registerBtn = page.getByRole('button', { name: 'Register' });
+        await expect(registerBtn).toBeDisabled();
+    });
+
+    test('Name input field validation', async ({page})=>{
+
+        const nameInput =  page.locator('#signupName');
+        const emptyNameError = page.getByText('Name required');
+        const invalidNameError = page.getByText('Name is invalid');
+        const nameLengthValidationError = page.getByText('Name has to be from 2 to 20 characters long');
```

Comment:
do not use locator based on text for errors, you cant guarantee this text is rendered in right place with this locator